BrainPortal/app/controllers/application_controller.rb

```diff
@@ -297,6 +298,21 @@ def disable_cookies_for_api
   # CBRAIN Messaging System Filters
   ########################################################################
 
+  # Count the number of active BackgroundActivities
+  def count_background_activities #:nodoc:
+    return unless current_user
+    return if     request.format.blank? || request.xhr?
```

Comment:
Not exactly. The next line would crash if `request.format` was nil, because `nil.to_sym` doesn't work. For sure we could rewrite the next line as `return unless request.format&.to_sym == :html` but the original code this was copied from came from older ruby that didn't have `&.`

Reply:
fair enough, even though I am a bit surprised that nil format exists, I thought that bug was fixed [long time ago](https://github.com/rails/rails/pull/8084).  Also I just realized that `request.xhr?` might be or become  different `request.format.xhr?`. Anyways I would merge two lines into 

```return if  request.format.blank? || ! request.format.html? || request.xhr?```

which would keep blank check near html check. and would avoid unless, which in English usually used account for abnormal situations. (So I do not mind